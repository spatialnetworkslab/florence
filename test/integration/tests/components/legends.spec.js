/// <reference types="Cypress" />

context('Assertions', () => {
  beforeEach(() => {
    cy.visit('/legends')
  })

  describe('Title Tests', () => {
    it('[render] legends rendered', () => {
      cy.get('.discrete-legend').should('have.length', 2)
      cy.get('.gradient-legend').should('have.length', 2)
    })

    it('[render] legends have title', () => {
      cy.get('g.discrete-legend > .label').should('have.length', 2)
      cy.get('g.gradient-legend > .label').should('have.length', 2)
      cy.get('g.gradient-legend > .label').should('contain', 'Test title 12345')
    })

    it('[render] discrete legend has rectangles', () => {
      cy.get('g.discrete-legend > g.rectangle-layer').should('have.length', 2)
      cy.get('g.discrete-legend > g.rectangle-layer > path.rectangle').should('have.length.greaterThan', 2)
    })

    it('[render] gradient legend has rectangles with gradients', () => {
      cy.get('g.gradient-legend > path.rectangle').should('have.length', 2)
      cy.get('g.gradient-legend > defs > linearGradient > stop').should('have.length.greaterThan', 2)
      cy.get('g.gradient-legend > defs > linearGradient > stop').should('have.css', 'stop-color')
      cy.get('g.gradient-legend > defs > linearGradient > stop').should('have.css', 'stop-opacity')
    })

    it('[render] legends have tick labels', () => {
      cy.get('g.discrete-legend > .label-layer > .label').should('have.length.greaterThan', 1)
      cy.get('g.gradient-legend > .label-layer > .label').should('have.length.greaterThan', 1)
    })
  })
})
