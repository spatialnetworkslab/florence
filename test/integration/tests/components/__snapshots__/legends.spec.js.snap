exports[`Assertions > Title Tests > [render] legends snapshot correctly #0`] = `
<svg width="700" height="800">
  <defs>
    <clipPath id="clip-sc0">
      <rect
        x="58.33333333333333"
        y="40"
        width="291.6666666666667"
        height="360"
      ></rect>
    </clipPath>
    <clipPath id="clip-sc0-data">
      <rect
        x="88.33333333333333"
        y="70"
        width="231.66666666666669"
        height="300"
        fill="white"
      ></rect>
    </clipPath>
  </defs>
  <g class="section" clip-path="url(#clip-sc0)">
    <g class="discrete-legend">
      <g class="rectangle-layer">
        <path
          class="rectangle"
          d="M200,90L216.66666666666666,90L216.66666666666666,60L200,60Z"
          fill="rgb(243, 0, 12)"
          stroke="none"
          stroke-width="2"
          fill-opacity="1"
          opacity="1"
        ></path>
        <path
          class="rectangle"
          d="M216.66666666666666,90L233.33333333333331,90L233.33333333333331,60L216.66666666666666,60Z"
          fill="rgb(202, 0, 53)"
          stroke="none"
          stroke-width="2"
          fill-opacity="1"
          opacity="1"
        ></path>
        <path
          class="rectangle"
          d="M233.33333333333331,90L249.99999999999997,90L249.99999999999997,60L233.33333333333331,60Z"
          fill="rgb(161, 0, 94)"
          stroke="none"
          stroke-width="2"
          fill-opacity="1"
          opacity="1"
        ></path>
        <path
          class="rectangle"
          d="M249.99999999999997,90L266.66666666666663,90L266.66666666666663,60L249.99999999999997,60Z"
          fill="rgb(120, 0, 135)"
          stroke="none"
          stroke-width="2"
          fill-opacity="1"
          opacity="1"
        ></path>
        <path
          class="rectangle"
          d="M266.66666666666663,90L283.3333333333333,90L283.3333333333333,60L266.66666666666663,60Z"
          fill="rgb(79, 0, 176)"
          stroke="none"
          stroke-width="2"
          fill-opacity="1"
          opacity="1"
        ></path>
        <path
          class="rectangle"
          d="M283.3333333333333,90L300,90L300,60L283.3333333333333,60Z"
          fill="rgb(38, 0, 217)"
          stroke="none"
          stroke-width="2"
          fill-opacity="1"
          opacity="1"
        ></path>
      </g>
      <g class="label-layer">
        <text
          class="label"
          y="100"
          fill="black"
          stroke="none"
          stroke-width="0"
          x="208.33333333333334"
          opacity="1"
          transform="
          rotate(0, 
          208.33333333333334, 
          100)
        "
          font-family="Helvetica"
          font-size="10px"
          font-weight="normal"
          text-anchor="middle"
          dominant-baseline="middle"
        >
          20
        </text>
        <text
          class="label"
          y="100"
          fill="black"
          stroke="none"
          stroke-width="0"
          x="225"
          opacity="1"
          transform="
          rotate(0, 
          225, 
          100)
        "
          font-family="Helvetica"
          font-size="10px"
          font-weight="normal"
          text-anchor="middle"
          dominant-baseline="middle"
        >
          40
        </text>
        <text
          class="label"
          y="100"
          fill="black"
          stroke="none"
          stroke-width="0"
          x="241.66666666666669"
          opacity="1"
          transform="
          rotate(0, 
          241.66666666666669, 
          100)
        "
          font-family="Helvetica"
          font-size="10px"
          font-weight="normal"
          text-anchor="middle"
          dominant-baseline="middle"
        >
          60
        </text>
        <text
          class="label"
          y="100"
          fill="black"
          stroke="none"
          stroke-width="0"
          x="258.3333333333333"
          opacity="1"
          transform="
          rotate(0, 
          258.3333333333333, 
          100)
        "
          font-family="Helvetica"
          font-size="10px"
          font-weight="normal"
          text-anchor="middle"
          dominant-baseline="middle"
        >
          80
        </text>
        <text
          class="label"
          y="100"
          fill="black"
          stroke="none"
          stroke-width="0"
          x="275"
          opacity="1"
          transform="
          rotate(0, 
          275, 
          100)
        "
          font-family="Helvetica"
          font-size="10px"
          font-weight="normal"
          text-anchor="middle"
          dominant-baseline="middle"
        >
          100
        </text>
        <text
          class="label"
          y="100"
          fill="black"
          stroke="none"
          stroke-width="0"
          x="291.6666666666667"
          opacity="1"
          transform="
          rotate(0, 
          291.6666666666667, 
          100)
        "
          font-family="Helvetica"
          font-size="10px"
          font-weight="normal"
          text-anchor="middle"
          dominant-baseline="middle"
        >
          120
        </text>
      </g>
      <text
        class="label"
        y="70"
        fill="black"
        stroke="none"
        stroke-width="0"
        x="170"
        opacity="1"
        transform="rotate(0, 170, 70)"
        font-family="Helvetica"
        font-size="12px"
        font-weight="bold"
        text-anchor="middle"
        dominant-baseline="hanging"
      >
        Legend
      </text>
    </g>
    <g class="gradient-legend">
      <defs>
        <linearGradient id="gradient0" x1="0%" y1="0%" x2="100%" y2="0%">
          <stop
            key="0"
            offset="8.333333333333332%"
            style="stop-color:rgb(243, 0, 12);stop-opacity:1"
          ></stop>
          <stop
            key="1"
            offset="25%"
            style="stop-color:rgb(202, 0, 53);stop-opacity:1"
          ></stop>
          <stop
            key="2"
            offset="41.666666666666664%"
            style="stop-color:rgb(161, 0, 94);stop-opacity:1"
          ></stop>
          <stop
            key="3"
            offset="58.33333333333333%"
            style="stop-color:rgb(120, 0, 135);stop-opacity:1"
          ></stop>
          <stop
            key="4"
            offset="75%"
            style="stop-color:rgb(79, 0, 176);stop-opacity:1"
          ></stop>
          <stop
            key="5"
            offset="91.66666666666666%"
            style="stop-color:rgb(38, 0, 217);stop-opacity:1"
          ></stop>
        </linearGradient>
      </defs>
      <path
        class="rectangle"
        d="M99.23693962531583,225.96631939980156L173.69961102056766,225.96631939980156L173.69961102056766,258.98318213693875L99.23693962531583,258.98318213693875Z"
        fill="url(#gradient0)"
        stroke="none"
        stroke-width="2"
        opacity="1"
      ></path>
      <g class="label-layer">
        <text
          class="label"
          y="268.98318213693875"
          fill="black"
          stroke="none"
          stroke-width="0"
          x="105.44216224158681"
          opacity="1"
          transform="
          rotate(0, 
          105.44216224158681, 
          268.98318213693875)
        "
          font-family="Helvetica"
          font-size="10px"
          font-weight="normal"
          text-anchor="middle"
          dominant-baseline="middle"
        >
          20
        </text>
        <text
          class="label"
          y="268.98318213693875"
          fill="black"
          stroke="none"
          stroke-width="0"
          x="117.85260747412879"
          opacity="1"
          transform="
          rotate(0, 
          117.85260747412879, 
          268.98318213693875)
        "
          font-family="Helvetica"
          font-size="10px"
          font-weight="normal"
          text-anchor="middle"
          dominant-baseline="middle"
        >
          40
        </text>
        <text
          class="label"
          y="268.98318213693875"
          fill="black"
          stroke="none"
          stroke-width="0"
          x="130.26305270667075"
          opacity="1"
          transform="
          rotate(0, 
          130.26305270667075, 
          268.98318213693875)
        "
          font-family="Helvetica"
          font-size="10px"
          font-weight="normal"
          text-anchor="middle"
          dominant-baseline="middle"
        >
          60
        </text>
        <text
          class="label"
          y="268.98318213693875"
          fill="black"
          stroke="none"
          stroke-width="0"
          x="142.67349793921272"
          opacity="1"
          transform="
          rotate(0, 
          142.67349793921272, 
          268.98318213693875)
        "
          font-family="Helvetica"
          font-size="10px"
          font-weight="normal"
          text-anchor="middle"
          dominant-baseline="middle"
        >
          80
        </text>
        <text
          class="label"
          y="268.98318213693875"
          fill="black"
          stroke="none"
          stroke-width="0"
          x="155.0839431717547"
          opacity="1"
          transform="
          rotate(0, 
          155.0839431717547, 
          268.98318213693875)
        "
          font-family="Helvetica"
          font-size="10px"
          font-weight="normal"
          text-anchor="middle"
          dominant-baseline="middle"
        >
          100
        </text>
        <text
          class="label"
          y="268.98318213693875"
          fill="black"
          stroke="none"
          stroke-width="0"
          x="167.49438840429667"
          opacity="1"
          transform="
          rotate(0, 
          167.49438840429667, 
          268.98318213693875)
        "
          font-family="Helvetica"
          font-size="10px"
          font-weight="normal"
          text-anchor="middle"
          dominant-baseline="middle"
        >
          120
        </text>
      </g>
      <text
        class="label"
        y="279.98880304931777"
        fill="black"
        stroke="none"
        stroke-width="0"
        x="136.46827532294174"
        opacity="1"
        transform="rotate(0, 136.46827532294174, 279.98880304931777)"
        font-family="Helvetica"
        font-size="12px"
        font-weight="bold"
        text-anchor="middle"
        dominant-baseline="hanging"
      >
        Legend
      </text>
    </g>
    <g class="gradient-legend">
      <defs>
        <linearGradient id="gradient1" x1="0%" y1="0%" x2="0%" y2="100%">
          <stop
            key="0"
            offset="0%"
            style="stop-color:purple;stop-opacity:0"
          ></stop>
          <stop
            key="1"
            offset="8.333333333333332%"
            style="stop-color:purple;stop-opacity:0.08333333333333333"
          ></stop>
          <stop
            key="2"
            offset="16.666666666666664%"
            style="stop-color:purple;stop-opacity:0.16666666666666666"
          ></stop>
          <stop
            key="3"
            offset="33.33333333333333%"
            style="stop-color:purple;stop-opacity:0.3333333333333333"
          ></stop>
          <stop
            key="4"
            offset="75%"
            style="stop-color:purple;stop-opacity:0.75"
          ></stop>
          <stop
            key="5"
            offset="100%"
            style="stop-color:purple;stop-opacity:1"
          ></stop>
        </linearGradient>
      </defs>
      <path
        class="rectangle"
        d="M296.3333333333333,193L313.7083333333333,193L313.7083333333333,283L296.3333333333333,283Z"
        fill="url(#gradient1)"
        stroke="none"
        stroke-width="2"
        opacity="1"
      ></path>
      <g class="label-layer">
        <text
          class="label"
          y="193"
          fill="black"
          stroke="none"
          stroke-width="0"
          x="286.3333333333333"
          opacity="1"
          transform="
          rotate(0, 
          286.3333333333333, 
          193)
        "
          font-family="Helvetica"
          font-size="10px"
          font-weight="normal"
          text-anchor="middle"
          dominant-baseline="middle"
        >
          0
        </text>
        <text
          class="label"
          y="200.5"
          fill="black"
          stroke="none"
          stroke-width="0"
          x="286.3333333333333"
          opacity="1"
          transform="
          rotate(0, 
          286.3333333333333, 
          200.5)
        "
          font-family="Helvetica"
          font-size="10px"
          font-weight="normal"
          text-anchor="middle"
          dominant-baseline="middle"
        >
          10
        </text>
        <text
          class="label"
          y="208"
          fill="black"
          stroke="none"
          stroke-width="0"
          x="286.3333333333333"
          opacity="1"
          transform="
          rotate(0, 
          286.3333333333333, 
          208)
        "
          font-family="Helvetica"
          font-size="10px"
          font-weight="normal"
          text-anchor="middle"
          dominant-baseline="middle"
        >
          20
        </text>
        <text
          class="label"
          y="223"
          fill="black"
          stroke="none"
          stroke-width="0"
          x="286.3333333333333"
          opacity="1"
          transform="
          rotate(0, 
          286.3333333333333, 
          223)
        "
          font-family="Helvetica"
          font-size="10px"
          font-weight="normal"
          text-anchor="middle"
          dominant-baseline="middle"
        >
          40
        </text>
        <text
          class="label"
          y="260.5"
          fill="black"
          stroke="none"
          stroke-width="0"
          x="286.3333333333333"
          opacity="1"
          transform="
          rotate(0, 
          286.3333333333333, 
          260.5)
        "
          font-family="Helvetica"
          font-size="10px"
          font-weight="normal"
          text-anchor="middle"
          dominant-baseline="middle"
        >
          90
        </text>
        <text
          class="label"
          y="283"
          fill="black"
          stroke="none"
          stroke-width="0"
          x="286.3333333333333"
          opacity="1"
          transform="
          rotate(0, 
          286.3333333333333, 
          283)
        "
          font-family="Helvetica"
          font-size="10px"
          font-weight="normal"
          text-anchor="middle"
          dominant-baseline="middle"
        >
          120
        </text>
      </g>
      <text
        class="label"
        y="235"
        fill="black"
        stroke="none"
        stroke-width="0"
        x="271.3333333333333"
        opacity="1"
        transform="rotate(-90, 271.3333333333333, 235)"
        font-family="Helvetica"
        font-size="12px"
        font-weight="bold"
        text-anchor="middle"
        dominant-baseline="hanging"
      >
        Test title 12345
      </text>
    </g>
  </g>
  <g class="discrete-legend">
    <g class="rectangle-layer">
      <path
        class="rectangle"
        d="M18,18L70.5,18L70.5,43L18,43Z"
        fill="#1b9e77"
        stroke="white"
        stroke-width="2"
        fill-opacity="1"
        opacity="1"
      ></path>
      <path
        class="rectangle"
        d="M18,43L70.5,43L70.5,68L18,68Z"
        fill="#d95f02"
        stroke="white"
        stroke-width="2"
        fill-opacity="1"
        opacity="1"
      ></path>
      <path
        class="rectangle"
        d="M18,68L70.5,68L70.5,93L18,93Z"
        fill="#7570b3"
        stroke="white"
        stroke-width="2"
        fill-opacity="1"
        opacity="1"
      ></path>
      <path
        class="rectangle"
        d="M18,93L70.5,93L70.5,118L18,118Z"
        fill="#e7298a"
        stroke="white"
        stroke-width="2"
        fill-opacity="1"
        opacity="1"
      ></path>
    </g>
    <g class="label-layer">
      <text
        class="label"
        y="30.5"
        fill="black"
        stroke="none"
        stroke-width="0"
        x="80.5"
        opacity="1"
        transform="
          rotate(0, 
          80.5, 
          30.5)
        "
        font-family="Helvetica"
        font-size="10px"
        font-weight="normal"
        text-anchor="start"
        dominant-baseline="middle"
      >
        apple
      </text>
      <text
        class="label"
        y="55.5"
        fill="black"
        stroke="none"
        stroke-width="0"
        x="80.5"
        opacity="1"
        transform="
          rotate(0, 
          80.5, 
          55.5)
        "
        font-family="Helvetica"
        font-size="10px"
        font-weight="normal"
        text-anchor="start"
        dominant-baseline="middle"
      >
        banana
      </text>
      <text
        class="label"
        y="80.5"
        fill="black"
        stroke="none"
        stroke-width="0"
        x="80.5"
        opacity="1"
        transform="
          rotate(0, 
          80.5, 
          80.5)
        "
        font-family="Helvetica"
        font-size="10px"
        font-weight="normal"
        text-anchor="start"
        dominant-baseline="middle"
      >
        orange
      </text>
      <text
        class="label"
        y="105.5"
        fill="black"
        stroke="none"
        stroke-width="0"
        x="80.5"
        opacity="1"
        transform="
          rotate(0, 
          80.5, 
          105.5)
        "
        font-family="Helvetica"
        font-size="10px"
        font-weight="normal"
        text-anchor="start"
        dominant-baseline="middle"
      >
        pomelo
      </text>
    </g>
    <text
      class="label"
      y="6"
      fill="black"
      stroke="none"
      stroke-width="0"
      x="45.5"
      opacity="1"
      transform="rotate(0, 45.5, 6)"
      font-family="Helvetica"
      font-size="12px"
      font-weight="bold"
      text-anchor="middle"
      dominant-baseline="hanging"
    >
      Legend
    </text>
  </g>
</svg>
`;
